File: src/DashboardUI/src/data-contracts/EstimateDetails.ts
import { CompensationRateUnits } from './CompensationRateUnits';
import { ControlLocationDetails } from './ControlLocationDetails';
import { LocationDetails } from './LocationDetails';

export interface EstimateDetails {
  id: string;
  compensationRateDollars: number;
  compensationRateUnits: Exclude<CompensationRateUnits, CompensationRateUnits.None> | undefined;
  estimatedCompensationDollars: number;
  cumulativeTotalEtInAcreFeet: number;
  cumulativeNetEtInAcreFeet: number | null;
  locations: LocationDetails[];
  controlLocation: ControlLocationDetails;
}

File: src/API/WesternStatesWater.WestDaat.Common/DataContracts/EstimateDetails.cs
namespace WesternStatesWater.WestDaat.Common.DataContracts;

public class EstimateDetails
{
    public Guid Id { get; set; }

    public int CompensationRateDollars { get; set; }

    public CompensationRateUnits CompensationRateUnits { get; set; }

    public int EstimatedCompensationDollars { get; set; }

    public double CumulativeTotalEtInAcreFeet { get; set; }

    public double? CumulativeNetEtInAcreFeet { get; set; }

    public LocationDetails[] Locations { get; set; } = null!;

    public ControlLocationDetails ControlLocation { get; set; } = null!;
}
